### PR TITLE
changes to run locally as lib

### PR DESCRIPTION
**Pull request by tarpdalton**
i have no idea what side effects these changes have, but this is what I changed to get things to run locally

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,14 +20,14 @@ ifeq ($(MIX_DEPS_PATH),)
 $(warning MIX_APP_PATH not set. Invoke via mix)
 endif
 
+CPP_PATH=$(MIX_DEPS_PATH)/pulsar/pulsar-client-cpp
```
Comment (tarpdalton) [inline Makefile:23]: i think cpp path needs to be defined before

File: Makefile [hunk 2]
```diff
@@ -20,14 +20,14 @@ ifeq ($(MIX_DEPS_PATH),)
...
+
 ifeq ($(shell uname),Darwin)     # Mac OS X
-PLATFORM_OPTIONS=-undefined dynamic_lookup -L$(CPP_PATH)/lib -lpulsar
+PLATFORM_OPTIONS=-undefined dynamic_lookup -L$(CPP_PATH)/lib -lpulsar -rpath $(CPP_PATH)/lib
```
Comment (tarpdalton) [inline Makefile:26]: this fixed the dynamic linking problem i had

```
13:37:30.784 [warn]  The on_load function for module Elixir.Neutron.PulsarNifs returned:
{:error, {:load_failed, 'Failed to load NIF library: \'dlopen(/Users/jamesdalton/git/blueshift-labs/schedulex/_build/dev/lib/neutron/priv/neutron_nif.so, 2): Library not loaded: @rpath/libpulsar.2.6.0.dylib\n  Referenced from: /Users/jamesdalton/git/blueshift-labs/schedulex/_build/dev/lib/neutron/priv/neutron_nif.so\n  Reason: image not found\''}}
```

File: Makefile [hunk 3]
```diff
@@ -39,9 +39,9 @@ priv:
 	mkdir -p $(MIX_APP_PATH)/priv/
 
 priv/neutron_nif.so: ./c_src/neutron_nif.c
-	  $(CC) $^ -shared -lei $(PLATFORM_OPTIONS) -fPIC -O3 -finline-functions -Wunused -Wall -Wpointer-arith -Wcast-align -Wcast-qual $(ERL_CFLAGS) $(ERTS_CFLAGS) $(ERL_LDFLAGS) -dynamiclib -pedantic -I/usr/local/ssl/include -L/usr/local/ssl/lib -I$(CPP_PATH)/include -o $(MIX_APP_PATH)/priv/neutron_nif.so
+	  $(CC) $^ -shared -lei $(PLATFORM_OPTIONS) -fPIC -O3 -finline-functions -Wunused -Wall -Wpointer-arith -Wcast-align -Wcast-qual $(ERL_CFLAGS) $(ERTS_CFLAGS) $(ERL_LDFLAGS) -dynamiclib -pedantic -I/usr/local/opt/openssl/include -L/usr/local/opt/openssl/lib -I$(CPP_PATH)/include -o $(MIX_APP_PATH)/priv/neutron_nif.so
```
Comment (tarpdalton) [inline Makefile]: might need to make this a dynamic var instead of hardcoded

File: build_deps.sh [hunk 4]
```diff
@@ -38,15 +38,15 @@ function BuildLib()
...
+            brew install openssl protobuf boost boost-python boost-python3 log4cxx jsoncpp
             export OPENSSL_ROOT_DIR=$(brew --prefix openssl)
             export OPENSSL_INCLUDE_DIR=$OPENSSL_ROOT_DIR/include
             export OPENSSL_SSL_LIBRARY=$OPENSSL_ROOT_DIR/lib
             export CPPFLAGS=-I$OPENSSL_ROOT_DIR/include
             export LDFLAGS=-L$OPENSSL_ROOT_DIR/lib
             cd pulsar-client-cpp
-            fail_check cmake . -DBUILD_TESTS=OFF
-            fail_check make pulsarShared -j$(nproc)
+            fail_check cmake . -DBUILD_TESTS=OFF -DBUILD_PYTHON_WRAPPER=OFF -DBoost_INCLUDE_DIRS=$(brew --prefix openssl)/include -DProtobuf_INCLUDE_DIR=$(brew --prefix protobuf)/include -DProtobuf_LIBRARIES=$(brew --prefix protobuf)/lib/libprotobuf.dylib
```
Comment (tarpdalton) [inline build_deps.sh]: took this from here https://github.com/Homebrew/homebrew-core/blob/master/Formula/libpulsar.rb

File: build_deps.sh [hunk 5]
```diff
@@ -38,15 +38,15 @@ function BuildLib()
...
+            fail_check make pulsarShared pulsarStatic -j$(nproc)
```
Comment (tarpdalton) [inline build_deps.sh]: do we need this for static subscriptions?

File: lib/neutron/pulsar_nifs.ex [hunk 6]
```diff
@@ -16,47 +16,47 @@ defmodule Neutron.PulsarNifs do
     :erlang.load_nif(directory, 0)
   end
 
-  def sync_produce(client_ref, topic, message, produce_opts) do
+  def sync_produce(_client_ref, _topic, _message, _produce_opts) do
```
Comment (tarpdalton) [inline lib/neutron/pulsar_nifs.ex:19]: added underscore since they are unused right now

File: Makefile (hunk 3 above)
Comment (v-kat) [reply Makefile]: This needs to be moved into mac and linux special-casing

File: build_deps.sh (hunk 5 above)
Comment (v-kat) [reply build_deps.sh]: This isn't what you think it is

File: Makefile (hunk 3 above)
Comment (tarpdalton) [reply Makefile]: ill just revert it. it just gives a warning

File: build_deps.sh [hunk 7]
```diff
@@ -38,14 +38,14 @@ function BuildLib()
 
   case $OS in
         Darwin)
-            brew install openssl protobuf boost boost-python log4cxx jsoncpp
+            brew install openssl protobuf boost boost-python boost-python3 log4cxx jsoncpp
```
Comment (v-kat) [inline build_deps.sh:41]: I don't think `boost-python3` is necessary all the time but it shouldn't cause any issues

**Pull request merged**